### DISCUSSION TITLE
Why is submitting tracked by the library

### DISCUSSION THREAD

**Discussion by JakeGinnivan** [airjp73/rvf#224] [Q&A] [answered]
Hey all,

I have just come across this library and am curious about the `useIsSubmitting` hook. Is this built on top of Remix's form functionality or does it rebuild it? At first glance I expected to just use the `useTransition` hook.

Just curious is all :)

**Comment by airjp73** [comment, marked as answer]
Hi!

It used to be a wrapper around `useTransition` until we added [async validation](https://www.remix-validated-form.io/async-validation). While validating, the form isn't really submitting from remix's perspective, but it is from the user's perspective. `useTransition` also turned out to be a big pain to use for automatically detecting submission in a general way that works no matter how many forms you have.

**Reply by JakeGinnivan** [reply]
Makes sense, thanks for the explanation!
